### PR TITLE
Snowflake source and destination: add postfix to hostname automatically

### LINKED ISSUE #2491: Snowflake destination quality of life improvement: default hostname postfix if not entered
## Tell us about the problem you're trying to solve
From https://airbytehq.slack.com/archives/C019WJFR1JT/p1615925659006000

If a user enters `<account_name>.ust-east-1` it would be nice if the connector autocompleted the host name.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368154212492) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sashaNeshcheret**
## What
We want to simplify inputting host to snowflake for user. by automatically adding immutable postfix ".aws.snowflakecomputing.com"
## How
Updated description in specs for host field, checked if host already contains ".aws.snowflakecomputing.com" and added it if missed

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by bazarnov — CHANGES_REQUESTED** [review]
You're also need to bump the Normalization version in these places:
1) `airbyte/integrations/bases/base-normalization/Dockerfile`
2) `airbyte/integrations/bases/base-normalization/snowflake.Dockerfile`
3) `airbyte/airbyte-workers/src/main/java/normalization/NormalizationRunnerFactory.java` (here the version should be the same as in `base-normalization/Dockerfile` after the bump)

After this you should `/test` & `/publish` the new Normalization image.
1) `/test connector=bases/base-normalization`
2) `/publish connector=bases/base-normalization`

File: airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSource.java [hunk 1]
```diff
@@ -63,6 +61,13 @@ public JsonNode toDatabaseConfig(final JsonNode config) {
     return Jsons.jsonNode(configBuilder.build());
   }
 
+  private String getHost(JsonNode config) {
+    String host = config.get("host").asText();
+    return host.endsWith(".aws.snowflakecomputing.com")
```
Comment (mkhokh-33) [inline airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSource.java:66]: we could make .aws.snowflakecomputing.com constant here

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java [hunk 2]
```diff
@@ -66,11 +67,19 @@ public static JdbcDatabase getDatabase(final JsonNode config) {
     return new DefaultJdbcDatabase(new SnowflakeConnectionSupplier(config));
   }
 
+  private static String getHost(JsonNode config) {
+    String host = config.get("host").asText();
+    return host.endsWith(".aws.snowflakecomputing.com")
```
Comment (mkhokh-33) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java:72]: .aws.snowflakecomputing.com make it constant

**Review by mkhokh-33 — CHANGES_REQUESTED** [review]

**Review by mkhokh-33 — APPROVED** [review]

**Review by tuliren — COMMENTED** [review]
I don't think we should merge this PR. 

This PR is assuming that the host is always on AWS, which is not true. According to this [doc](https://docs.snowflake.com/en/user-guide/intro-regions.html), Snowflake can be hosted on all three major cloud providers.

Actually Snowflake JDBC host can be identified without the cloud or region already. See [here](https://docs.snowflake.com/en/user-guide/jdbc-configure.html#syntax) for details. You only need to specify a host URL like this:

> <account_identifier>.snowflakecomputing.com

which is already pretty simple.

**Comment by bazarnov** [comment]
> This PR is assuming that the host is always on AWS, which is not true. According to this [doc](https://docs.snowflake.com/en/user-guide/intro-regions.html), Snowflake can be hosted on all three major cloud providers.

As the reviewer of this PR, I completely agree. 
What is the source of truth here, why the AWS is the only option?

**Comment by bazarnov** [comment]
> > This PR is assuming that the host is always on AWS, which is not true. According to this [doc](https://docs.snowflake.com/en/user-guide/intro-regions.html), Snowflake can be hosted on all three major cloud providers.
> 
> As the reviewer of this PR, I completely agree. What is the source of truth here, why the AWS is the only option?

Hi, what about this?

**Pull request closed without merging**